tests/portal/test_registration.py

```diff
@@ -0,0 +1,32 @@
+from fauxfactory import gen_string
+import unittest
+from insights.ui.login import LoginUI
+from insights.vm_provisioning import VirtualMachine
+
+
+class Register(unittest.TestCase):
+    def setUp(self):
+        self.login = LoginUI()
+        self.vm = VirtualMachine()
+        self.vm_name = 'vm_{0}'.format(gen_string('alpha', 6))
+        self.vm.create_openstack_instance(instance_name=self.vm_name, image_name='RHEL-7.1-x86_64',
+                                            flavor_name='Tiny', key_name='key1', pool_name='public')
+
+    def test_vm(self):
+        print self.vm
+        self.vm.rhsm_register(distro='rhel7')
+        self.vm.register_to_insights()
+        self.host_name = self.vm.hostname.rstrip('\n')
+        self.login.login_to_portal()
+        self.result_text = self.login.register_system(self.host_name)
+        self.assertIn(self.host_name, self.result_text)
+
+    def tearDown(self):
+        self.vm.unregister_from_rhi()
+        self.vm.unregister_from_RHSM()
+        self.vm.delete_openstack_instance(self.vm_name)
+        self.login.teardown()
+
+
+if __name__ == "__main__":
```

Comment:
the test runner shall be py.test and not unittest. execute your tests using "py.test -s -v tests/portal/test*.py".
I dont think we require to add this block of code.